src/workerd/io/actor-cache.c++

```diff
@@ -2781,6 +2812,9 @@ kj::Promise<void> ActorCache::flushImplDeleteAll(uint retryCount) {
 
     // Now we must flush any writes that happened after the deleteAll(). (If there are none, this
     // will complete quickly.)
+    // TODO(soon) This will use the write options for the deleteAll() even if the options for future
```

Comment:
Interesting, what makes this `flushImplDeleteAll()` case different from any other case where `allowUnconfirmed` is set on one write, but not the next, and then they're flushed together?  I thought that was all handled in `ensureFlushScheduled()` and I'm not seeing what would make this case different?

Reply:
The big difference is that this flush recursion is in the same promise chain as the original output gate wrapper. Imagine you have two sets of operations in different js tasks, normally they get separate flush promises with their own output gate locks. With this follow on `flushImpl()`, we still have two separate promises/flushes but the we handle the work for the second one in the first one. The second flush noops instead. I think this also happens when we have to retry failures?

I think it mostly matters when the deleteAll isn't allowUnconfirmed but the following operations are. In that case, we'll hold the output lock longer than the user requested. The reverse doesn't really matter because we're still holding the output lock as long as promised (assuming the user actually understood what was going on).

Reply:
I suppose, yeah. To clarify for my own sake -- it's already the case that if you do a `put` or normal `delete`  without `allowUnconfirmed` followed by another one with `allowUnconfirmed` before the flush has started that the output gate will apply to both. The difference is just that those operations are flushed atomically with each other whereas a deleteAll happens strictly before any follow-up operations.

So I think you're right that this is pessimistic but it seems like a pretty minor case given how infrequently used `deleteAll` is and that it requires mixing in `allowUnconfirmed` on only some operations.